traiders/backend/api/views/search.py

```diff
@@ -0,0 +1,100 @@
+
+from rest_framework.viewsets import GenericViewSet
+from rest_framework import mixins, status
+from django.db.models import Q
+from rest_framework.response import Response
+from django_filters.rest_framework import DjangoFilterBackend
+from rest_framework import serializers
+import requests as rq
+import logging
+
+from ..models import *
+from ..serializers import *
+
+MAX_ITEMS = 5
+logger = logging.getLogger(__name__)
+
+
+class SearchViewSet(GenericViewSet):
+    """
+    View searching a given string
+    """
+    filter_backends = [DjangoFilterBackend]
+
+    class serializer_class(serializers.Serializer):  # this is just for schema generation, not used
+        base_equipment = EquipmentSerializer()
+        target_equipment = EquipmentSerializer()
+        success_rate = serializers.FloatField()
+        prediction_count = serializers.IntegerField()
+        user = UserSerializer()
+
+    @staticmethod
+    def find_closest(keyword, top_k=10):
+        words = keyword.split(' ')
+        kws = []
+        for word in words:
+            kws.append((1, word))
+        keyword = keyword.replace(' ', '+')
+        res = rq.get(f'https://api.datamuse.com/words?ml={keyword}&max={top_k}')
```

Comment:
You can use `params` keyword when giving url parameters. This way you don't need to do the urlencoding (' '->+) yourself.
```python
rq.get('https://api.datamuse.com/words', params={'wl': 'us dollars', 'max': 5})
```